packages/lit-dev-content/site/blog/2022-02-07-eleventy.md

```diff
@@ -0,0 +1,181 @@
+on to learn more about what it does, how it works, and why you might want to use
+it.
+
+## What does it do?
+
+After you've added `@lit-labs/eleventy-plugin-lit` to your Eleventy config, then
+whenever you write a custom element tag in a markdown or HTML file, the initial
+state of that component's shadow DOM and styles will be rendered directly into
+the static HTML.
+
+For example, given a markdown file `hello.md`:
+
+```html
+# Greetings
+
+<demo-greeter name="World"></demo-greeter>
+```
+
+And a component definition `js/demo-greeter.js`:
+
+```js
+import {LitElement, html, css} from 'lit';
+
+class DemoGreeter extends LitElement {
+  static styles = css`
+    b { color: red; }
+  `;
+
+  static properties = {
+    name: {},
+  };
+
+  render() {
+    return html`Hello <b>${this.name}</b>!`;
+  }
+}
+customElements.define('demo-greeter', DemoGreeter);
+```
+
+Then Eleventy will produce an HTML file `greeting/index.html` like this:
+
+```html
+<h1>Greetings</h1>
+
+<demo-greeter name="World">
+  <template shadowroot="open">
+    <style>
+      b { color: red; }
+    </style>
+    Hello <b>World</b>!
+  </template>
+</demo-greeter>
+```
+
+This HTML allows the browser to paint the initial state of the component's DOM
+and styles, preserving all of the encapsulation boundaries of web components,
+even before a single line of JavaScript has downloaded!
+
+Later, when the JavaScript definition of `<demo-greeter>` has finally loaded,
+the pre-rendered component is upgraded to its fully interactive JavaScript
+implementation through a process called re-hydration.
+
+## Why should I use it?
+
+Pre-rendering components is an effective method for reducing the time it takes
+between a page starting to load, and the content becoming visible — a
+measurement known as the *time to first paint (FP)* or *time to first contentful
+paint (FCP)*. This is an important factor that contributes to the responsiveness
+of your site, particularly for users with slower connections and devices.
+
+By transmitting the pre-rendered initial state of your components using
+declarative HTML, the browser is able to paint the initial view much faster than
+it takes to download, parse, and evaluate the JavaScript implementations of
+those components.
+
+For a deeper dive into the benefits of pre-rendering, check out *[Rendering on
+the Web
+(developers.google.com)](https://developers.google.com/web/updates/2019/02/rendering-on-the-web)*.
+
+## How does it work?
+
+### Lit SSR
+
+The plugin uses `@lit-labs/ssr` under the hood, which works by creating an
+instance of each component in a lightweight browser-like environment within the
+Eleventy Node process, and evaluating the element's template using a special
+version of Lit's `render` function. This special `render` function behaves much
+like the `render` function that Lit uses in the browser, except that it emits
+stream of strings, instead of writing to the DOM.
```

Comment:
Nit: "**a** stream" or maybe "stream**s** of strings"? If it's one stream then I think it needs the added "a" before it.

Reply:
Oops yep thanks. Done!